python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -311,6 +312,7 @@ def get_required_multi_asset_keys(self, asset_key: AssetKey) -> AbstractSet[Asse
             return self._required_multi_asset_sets_by_key[asset_key]
         return set()
 
+    @cached_method
```

Comment:
for very big asset graphs, toposorting can take a few seconds, figured there wasn't much harm in caching this as we do call this method a few times